src/db.c

```diff
@@ -24,13 +24,16 @@
 #include <signal.h>
 #include <ctype.h>
 #include "bio.h"
+#include "keymeta.h"
 
 /*-----------------------------------------------------------------------------
  * C-level DB API
  *----------------------------------------------------------------------------*/
 
 static_assert(MAX_KEYSIZES_TYPES == OBJ_TYPE_BASIC_MAX, "Must be equal");
 
+const KeyMetaSpec KEY_METADATA_EMPTY = { 0 };
```

Comment:
what about moving this into dbAdd and dbAddInternal? It's only used by them two.
Reducing its valid domain feels more reasonable.